src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -1579,9 +1579,16 @@ memgraph::storage::PropertyValue ToPropertyValue(const mgp_value &value) {
 
 mgp_error mgp_vertex_set_property(struct mgp_vertex *v, const char *property_name, mgp_value *property_value) {
   return WrapExceptions([=] {
+    if (v->graph->ctx && v->graph->ctx->auth_checker &&
+        !v->graph->ctx->auth_checker->Accept(*v->graph->ctx->db_accessor, v->impl, v->graph->view,
+                                             memgraph::query::AuthQuery::FineGrainedPrivilege::UPDATE)) {
+      return;
+    }
+
```

Comment:
I would put this after the mutability check, but it is not an important change. My reasoning is using a "write" operation in a read procedure is a "bigger problem" than not having the privilege to modify a certain vertex.

And this will also result in an error in the future, right?

Reply:
> I would put this after the mutability check, but it is not an important change. My reasoning is using a "write" operation in a read procedure is a "bigger problem" than not having the privilege to modify a certain vertex.

I've placed the check here as a way of sort of protecting the information about the entity. We can discuss a bit more if You'd like

> And this will also result in an error in the future, right?

Yes this is addressed in the 3rd PR